packages/@tailwindcss-cli/src/utils/resolve.ts

```diff
@@ -1,3 +1,32 @@
+import EnhancedResolve from 'enhanced-resolve'
+import fs from 'node:fs'
 import { createRequire } from 'node:module'
 
-export const resolve = createRequire(import.meta.url).resolve
+const localResolve = createRequire(import.meta.url).resolve
+export function resolve(id: string) {
+  if (typeof globalThis.__tw_resolve === 'function') {
+    let resolved = globalThis.__tw_resolve(id)
+    if (resolved) {
+      return resolved
+    }
+  }
+  return localResolve(id)
+}
+
+const resolver = EnhancedResolve.ResolverFactory.createResolver({
```

Comment:
It's probably not feature-ful enough, but you might be able to instead use `Bun.resolveSync` here. `Bun.resolveSync` supports a 2nd argument for the path.

You can see some tests for it here:
https://github.com/oven-sh/bun/blob/6603133732892227f0a32e3f19b343fd1e59a604/test/js/bun/resolve/import-meta.test.js#L133-L154

Reply:
Un fortunately I don't think we can because we need support for the `style` field and `style` conditions in `exports`. Good to know about this though — perhaps it could be usable elsewhere.